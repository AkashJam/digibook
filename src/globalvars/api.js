const path = "http://10.0.2.2:3000";

const API = {
  // user
  login: async (user) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/user/login`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(user),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
  register: async (user) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/user/register`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(user),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
  updateUser: async (userdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/user/login`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(userdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },

  // groups
  getGroups: async (userID) => {
    //GET request
    const response = await fetch(`${path}/group/${userID}`, {
      method: "GET",
      //Request Type
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        //Success
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        //Error
        return error;
      });
    return response;
  },
  createGroup: async (groupdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/group/add`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(groupdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
  updateGroup: async (groupdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/group/update`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(groupdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },

  // Tasks
  getTasks: async (userID) => {
    //GET request
    const response = await fetch(`${path}/task/${userID}`, {
      method: "GET",
      //Request Type
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        //Success
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        //Error
        return error;
      });
    return response;
  },
  createTask: async (taskdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/task/add`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(taskdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
  updateTask: async (taskdata) => {
    //POST json
    //making data to send on server
    // const data = { username: username, password: password };
    //POST request
    const response = await fetch(`${path}/task/update`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(taskdata),
    })
      .then((response) => response.json())
      //If response is in json then in success
      .then((responseJson) => {
        return responseJson;
      })
      //If response is not in json then in error
      .catch((error) => {
        return error;
      });
    return response;
  },
};

export default API;
