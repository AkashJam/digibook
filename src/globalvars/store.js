import React, { useReducer } from "react";
import { toastr } from "./";
import AsyncStorage from "@react-native-async-storage/async-storage";

const user = {
  id: 0,
  username: "",
  displayName: null,
  range: 0.75,
  collaborators: [],
  groups: [],
  activities: [],
  notifications: [],
  locations: {},
};

const storeUserLog = (data) =>
  AsyncStorage.setItem(`DN_userlog`, JSON.stringify(data));

const reducer = (state, action) => {
  let log = [];
  let obj = {};
  switch (action.type) {
    //users

    case "set_user":
      // console.log(action.locations);
      obj = {
        id: action.user.id,
        username: action.user.username,
        displayName: action.user.displayName,
        range: action.user.range,
        collaborators: action.user.collaborators,
        groups: action.user.groups,
        activities: action.user.activities,
        notifications: action.notifications,
        locations: action.locations,
      };
      storeUserLog(obj);
      toastr(
        `Welcome ${
          action.user.displayName !== null
            ? action.user.displayName
            : action.user.username
        }`
      );
      return obj;

    case "update_display_name":
      // obj = await API.updateUser({
      //   id: state.id,
      //   task: { display_name: action.displayName },
      // });
      // if (obj.code == 200) {
      obj = {
        id: state.id,
        username: state.username,
        displayName: action.displayName,
        range: state.range,
        collaborators: action.collaborators,
        groups: state.groups,
        activities: state.activities,
        notifications: state.notifications,
        locations: state.locations,
      };
      storeUserLog(obj);
      toastr("Display name updated.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    case "update_user_range":
      // obj = await API.updateUser({
      //   id: state.id,
      //   user: { range: action.range },
      // });
      // if (obj.code == 200) {
      obj = {
        id: state.id,
        username: state.username,
        displayName: state.displayName,
        range: action.range,
        collaborators: state.collaborators,
        groups: state.groups,
        activities: state.activities,
        notifications: state.notifications,
        locations: state.locations,
      };
      storeUserLog(obj);
      toastr("Notification range updated.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    //activities
    case "add_task":
      const newTask = {
        id: action.id,
        description: action.description,
        group_id: action.group_id,
        notify: false,
        location: null,
        datetime: action.datetime !== undefined ? action.datetime : null,
        completed: false,
        createdOn: new Date().toString(),
        modifiedOn: null,
      };
      log = [newTask, ...state.activities];
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Task added.");
      return obj;

    case "rename_task":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            description: action.description,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Task renamed.");
      return obj;

    case "delete_task":
      log = state.activities.filter((activity) => activity.id !== action.id);
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Task deleted.");
      return obj;

    case "change_group":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            group_id: action.group_id,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Group changed.");
      return obj;

    case "change_datetime":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            datetime: action.datetime,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Date and time changed.");
      return obj;

    case "change_location":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            location: action.location,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr("Location changed.");
      return obj;

    case "toggle_notification":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            notify: action.notify,
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr(action.notify ? "Notification On." : "Notification Off.");
      return obj;

    case "toggle_completion":
      log = state.activities.map((activity) => {
        if (activity.id === action.id) {
          return {
            ...activity,
            completed: action.completed,
            datetime: new Date().toString(),
            modifiedOn: new Date().toString(),
          };
        } else {
          return activity;
        }
      });
      obj = { ...state, activities: log };
      storeUserLog(obj);
      toastr(
        action.completed ? "Task completed." : "Task no longer completed."
      );
      return obj;

    // groups
    case "add_group":
      let newGroup = {
        id: action.id,
        name: action.name,
        createdOn: new Date().toString(),
        modifiedOn: null,
      };
      log = [newGroup, ...state.groups];
      obj = { ...state, groups: log };
      storeUserLog(obj);
      toastr("Group added.");
      return obj;

    case "rename_group":
      // obj = await API.updateGroup({
      //   id: action.id,
      //   group: { name: action.name },
      // });
      // if (obj.code == 200) {
      log = state.groups.map((group) => {
        if (group.id === action.id) {
          return {
            ...group,
            name: action.name,
            modifiedOn: new Date().toString(),
          };
        } else {
          return group;
        }
      });
      obj = { ...state, groups: log };
      storeUserLog(obj);
      toastr("Group renamed.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    case "delete_group":
      // obj = await API.updateGroup({
      //   id: action.id,
      //   group: { active: false },
      // });
      // if (obj.code == 200) {
      log = state.groups.filter((group) => group.id !== action.id);
      obj = { ...state, groups: log };
      storeUserLog(obj);
      toastr("Group deleted.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    //notifications
    case "update_notifications":
      // obj = await API.updateUser({
      //   id: state.id,
      //   task: { display_name: action.displayName },
      // });
      // if (obj.code == 200) {
      obj = {
        id: state.id,
        username: state.username,
        displayName: state.displayName,
        range: state.range,
        collaborators: state.collaborators,
        groups: state.groups,
        activities: state.activities,
        notifications: action.notifications,
        locations: state.locations,
      };
      // storeUserLog(obj);
      // toastr("Display name updated.");
      return obj;
    // } else {
    //   toastr("Error connecting to server.", toastTime);
    //   return state;
    // }

    // locations
    case "set_location":
      obj = state.locations ? state.locations : {};
      if (
        !state.locations ||
        state.locations.lat === null ||
        state.locations.lon === null
      ) {
        obj.lat = action.latitude;
        obj.lon = action.longitude;
      } else if (
        Math.abs(obj.latitude - action.latitude) > 0.01 &&
        Math.abs(obj.longitude - action.longitude) > 0.01
      ) {
        Object.keys(obj).forEach((e) => {
          if (e !== "lat" && e !== "lon") obj[e] = [];
        });
        obj.lat = action.latitude;
        obj.lon = action.longitude;
      }
      obj[`${action.use}`] = action.location;
      storeUserLog({ ...state, locations: obj });
      return { ...state, locations: obj };

    default:
      return state;
  }
};

const UserContext = React.createContext({
  state: user,
  dispatch: () => null,
});

const UserProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, user);

  return (
    <UserContext.Provider value={[state, dispatch]}>
      {children}
    </UserContext.Provider>
  );
};

export { UserContext, UserProvider };
