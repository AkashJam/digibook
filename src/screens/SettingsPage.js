import React, { useState, useEffect, useRef } from "react";
import { Slider } from "@miblanchard/react-native-slider";
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  TouchableOpacity,
  TextInput,
} from "react-native";
import { Header } from "../components";
import { COLORS, FONTS, PAGE, PAGEHEAD, SIZES } from "../constants";
import { FontAwesome } from "@expo/vector-icons";
import { UserContext, toastr, API } from "../globalvars";

export default function SettingsPage() {
  const [state, dispatch] = React.useContext(UserContext);
  const [name, setName] = useState(state.displayName);
  const [value, setValue] = useState(state.range);
  const [password, setPassword] = useState("");
  const [text, setText] = useState("");
  const [change, setChange] = useState(false);

  useEffect(() => {
    if (
      ((name !== state.displayName && name !== "") || value !== state.range) &&
      (password === "" || password.length > 7)
    ) {
      setText("");
      setChange(true);
    } else setChange(false);
    if (name !== state.displayName && name === "")
      setText("Display name cannot be left empty");
    if (password !== "" && password.length <= 7)
      setText("Password needs to be atleast 8 characters");
  }, [name, password, value]);

  const editUserConfig = async () => {
    if (name !== state.displayName && name !== "") {
      const data = await API.updateUser({
        id: state.id,
        user: { display_name: name },
      });
      if (data.code === 200) {
        dispatch({
          type: "update_display_name",
          displayName: name,
        });
        console.log("name updated");
      }
    }
    if (value !== state.range) {
      const data = await API.updateUser({
        id: state.id,
        user: { range: value },
      });
      if (data.code === 200) {
        dispatch({
          type: "update_user_range",
          range: value,
        });
        console.log("range updated");
      }
    }
    if (password.length > 7) {
      const data = await API.updateUser({
        id: state.id,
        user: { password: password },
      });
      if (data.code === 200) {
        toastr("Password succcesfully updated");
      }
    }
    // toastr("User config updated")
  };

  return (
    <>
      <Header screen={"Settings"} />
      <View style={PAGE}>
        <View
          style={{
            paddingTop: SIZES.padding,
            width: "100%",
            alignItems: "center",
          }}
        >
          <FontAwesome
            name="user-circle"
            size={Dimensions.get("window").height / 4.5}
            color={COLORS.secondary}
          />
          <Text style={{ ...PAGEHEAD, color: COLORS.secondary }}>
            {state.username}
          </Text>
        </View>
        <View
          style={{
            // flex: 1,
            margin: SIZES.padding,
            backgroundColor: COLORS.primary,
            borderRadius: SIZES.borderRadius,
            overflow: "hidden",
          }}
        >
          <View style={{ flexDirection: "row" }}>
            <Text
              style={{
                ...styles.option,
                minWidth: 143,
              }}
            >
              Display name
            </Text>
            <TextInput
              placeholder="Enter your name"
              style={{
                flex: 1,
                padding: SIZES.padding,
                backgroundColor: COLORS.accent,
                ...FONTS.p_regular,
              }}
              value={name}
              onChangeText={(text) => setName(text)}
              // onFocus={() => setName(false)}
              // onBlur={() => setName(true)}
            />
          </View>
          <View
            style={{
              flexDirection: "row",
              paddingVertical: SIZES.margin,
              // backgroundColor: COLORS.primary,
            }}
          >
            <Text
              style={{
                ...styles.option,
                minWidth: 143,
              }}
            >
              Password
            </Text>
            <TextInput
              placeholder="Enter new password"
              style={{
                flex: 1,
                paddingHorizontal: SIZES.padding,
                backgroundColor: COLORS.accent,
                ...FONTS.p_regular,
                fontSize: 16,
              }}
              value={password}
              onChangeText={(text) => setPassword(text)}
              secureTextEntry={true}
            />
          </View>
          <View
            style={{
              flexDirection: "row",
              // paddingVertical: SIZES.margin,
            }}
          >
            <Text
              style={{
                ...styles.option,
                minWidth: 143,
              }}
            >
              Range: {Math.round(value * 1000)}m
            </Text>
            <View
              style={{
                flex: 1,
                paddingHorizontal: SIZES.padding,
                paddingVertical: "1%",
                backgroundColor: COLORS.accent,
              }}
            >
              <Slider
                step={0.1}
                minimumValue={0.1}
                maximumValue={1}
                value={value}
                onValueChange={(value) => setValue(value[0])}
                thumbTintColor={COLORS.primary}
                // animateTransitions={true}
              />
            </View>
          </View>
          {/* <Text
            style={{
              ...styles.option,
              marginVertical: SIZES.margin,
            }}
          >
            View Collaborators
          </Text> */}
          <Text style={{ ...styles.option, marginTop: SIZES.margin }}>
            Logout
          </Text>
        </View>
        {change && (
          <TouchableOpacity onPress={editUserConfig} style={styles.button}>
            <Text style={{ ...FONTS.h2_bold, color: COLORS.secondary }}>
              Confirm
            </Text>
          </TouchableOpacity>
        )}
        {text !== "" && (
          <Text style={{ ...PAGEHEAD, textAlign: "center" }}>{text}</Text>
        )}
      </View>
    </>
  );
}

const styles = StyleSheet.create({
  option: {
    ...FONTS.p_regular,
    color: COLORS.accent,
    backgroundColor: COLORS.secondary,
    padding: SIZES.padding,
  },
  button: {
    marginVertical: SIZES.padding,
    marginHorizontal: "25%",
    borderRadius: SIZES.borderRadius,
    padding: SIZES.margin,
    backgroundColor: COLORS.accent,
    alignItems: "center",
  },
});
