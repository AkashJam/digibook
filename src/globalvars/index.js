import API from "./api";
import { UserContext, UserProvider, toastr } from "./store";

export { toastr, UserContext, UserProvider, API };
