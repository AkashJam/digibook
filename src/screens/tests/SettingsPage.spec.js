import SettingsPage from "../SettingsPage";
import renderer from "react-test-renderer";
import { UserProvider } from "../../globalvars";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

// mock navigation
const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  return {
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

jest.mock("@expo/vector-icons", () => {
  const { View } = require("react-native");
  return {
    MaterialCommunityIcons: View,
    FontAwesome: View,
    Ionicons: View,
  };
});

jest.mock("react-native-root-toast", () => {
  const { View } = require("react-native");
  return {
    Toast: View,
  };
});

jest.mock("react-native-maps", () => {
  const { View } = require("react-native");
  return {
    MapView: View,
    Marker: View,
  };
});

jest.mock("react-native-gesture-handler", () => {
  const { View } = require("react-native");
  return {
    FlatList: View,
  };
});

describe("SettingsPage", () => {
  it("renders without crashing", () => {
    const rendered = renderer
      .create(
        <UserProvider>
          <SettingsPage />
        </UserProvider>
      )
      .toJSON();
    expect(rendered).toBeTruthy();
  });

  it("display name renders", () => {
    const { queryByText } = render(
      <UserProvider>
        <SettingsPage />
      </UserProvider>
    );
    expect(queryByText("tester")).not.toBeNull();
  });

  it("confirm button renders", () => {
    const { queryByText, getByPlaceholderText } = render(
      <UserProvider>
        <SettingsPage />
      </UserProvider>
    );
    fireEvent.changeText(getByPlaceholderText("Enter your name"), "test");
    expect(queryByText("Confirm")).not.toBeNull();
  });

  it("confirm button does not render when invalid input", () => {
    const { queryByText, getByPlaceholderText } = render(
      <UserProvider>
        <SettingsPage />
      </UserProvider>
    );
    fireEvent.changeText(getByPlaceholderText("Enter your name"), "");
    expect(queryByText("Confirm")).toBeNull();
  });

  it("test against snapshot", () => {
    const tree = renderer
      .create(
        <UserProvider>
          <SettingsPage />
        </UserProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
