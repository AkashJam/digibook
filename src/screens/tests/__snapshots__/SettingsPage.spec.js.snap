// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPage test against snapshot 1`] = `
Array [
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-u5wzg0 r-display-6koalj r-flexDirection-18u37iz r-height-1ph75f1 r-paddingHorizontal-1j3t67a r-paddingTop-sg165l r-position-u8s1d r-width-13qz1uu r-zIndex-136ojw6"
  >
    <div
      className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
      onClick={[Function]}
      onContextMenu={[Function]}
      onKeyDown={[Function]}
      role="menu"
      style={
        Object {
          "WebkitTransitionDuration": "0s",
          "transitionDuration": "0s",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-view-1dbjc4n"
      />
    </div>
    <div
      className="css-text-901oao"
      dir="auto"
      style={
        Object {
          "color": "rgba(54,54,53,1.00)",
          "fontSize": "18px",
          "marginLeft": "10px",
        }
      }
    >
      Settings
    </div>
  </div>,
  <div
    className="css-view-1dbjc4n r-backgroundColor-u5wzg0"
    style={
      Object {
        "height": "100%",
        "paddingLeft": "4%",
        "paddingRight": "4%",
        "paddingTop": "90px",
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-width-13qz1uu"
      style={
        Object {
          "paddingTop": "4%",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
      />
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(54,54,53,1.00)",
            "fontFamily": "Caviar_Dreams_Bold",
            "fontSize": "24px",
            "paddingBottom": "4%",
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "paddingTop": "4%",
          }
        }
      >
        tester
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-backgroundColor-u5wzg0 r-borderRadius-1ylenci"
      style={
        Object {
          "marginBottom": "4%",
          "marginLeft": "4%",
          "marginRight": "4%",
          "marginTop": "4%",
          "overflowX": "hidden",
          "overflowY": "hidden",
        }
      }
    >
      <div
        className="css-view-1dbjc4n r-flexDirection-18u37iz"
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "minWidth": "143px",
            }
          }
        >
          Display name
        </div>
        <input
          autoCapitalize="sentences"
          autoComplete="on"
          autoCorrect="on"
          className="css-textinput-11aywtz r-backgroundColor-zes02y"
          dir="auto"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onSelect={[Function]}
          placeholder="Enter your name"
          readOnly={false}
          spellCheck={true}
          style={
            Object {
              "WebkitFlex": 1,
              "flex": 1,
              "fontFamily": "Fredoka_Regular",
              "fontSize": "18px",
              "msFlex": "1 1 0%",
              "paddingBottom": "4%",
              "paddingLeft": "4%",
              "paddingRight": "4%",
              "paddingTop": "4%",
            }
          }
          type="text"
          value={null}
        />
      </div>
      <div
        className="css-view-1dbjc4n r-flexDirection-18u37iz"
        style={
          Object {
            "paddingBottom": "2%",
            "paddingTop": "2%",
          }
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "minWidth": "143px",
            }
          }
        >
          Password
        </div>
        <input
          autoCapitalize="sentences"
          autoComplete="on"
          autoCorrect="on"
          className="css-textinput-11aywtz r-backgroundColor-zes02y"
          dir="auto"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onSelect={[Function]}
          placeholder="Enter new password"
          readOnly={false}
          spellCheck={true}
          style={
            Object {
              "WebkitFlex": 1,
              "flex": 1,
              "fontFamily": "Fredoka_Regular",
              "fontSize": "16px",
              "msFlex": "1 1 0%",
              "paddingLeft": "4%",
              "paddingRight": "4%",
            }
          }
          type="password"
          value=""
        />
      </div>
      <div
        className="css-view-1dbjc4n r-flexDirection-18u37iz"
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "minWidth": "143px",
            }
          }
        >
          Range: 
          750
          m
        </div>
        <div
          className="css-view-1dbjc4n r-backgroundColor-zes02y"
          style={
            Object {
              "WebkitFlex": 1,
              "flex": 1,
              "msFlex": "1 1 0%",
              "paddingBottom": "1%",
              "paddingLeft": "4%",
              "paddingRight": "4%",
              "paddingTop": "1%",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitBoxPack": "center",
                "WebkitJustifyContent": "center",
                "height": "40px",
                "justifyContent": "center",
                "msFlexPack": "center",
              }
            }
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "backgroundColor": "rgba(179,179,179,1.00)",
                  "borderBottomLeftRadius": "2px",
                  "borderBottomRightRadius": "2px",
                  "borderTopLeftRadius": "2px",
                  "borderTopRightRadius": "2px",
                  "height": "4px",
                }
              }
            />
            <div
              className="css-view-1dbjc4n r-position-u8s1d"
              id="animatedComponent"
              style={
                Object {
                  "backgroundColor": "rgba(63,63,63,1.00)",
                  "borderBottomLeftRadius": "2px",
                  "borderBottomRightRadius": "2px",
                  "borderTopLeftRadius": "2px",
                  "borderTopRightRadius": "2px",
                  "height": "4px",
                  "left": "0px",
                  "opacity": 0,
                  "width": "0px",
                }
              }
            />
            <div
              className="css-view-1dbjc4n r-backgroundColor-u5wzg0 r-position-u8s1d"
              id="animatedComponent"
              style={
                Object {
                  "WebkitTransform": "translateX(0px) translateY(0px)",
                  "borderBottomLeftRadius": "10px",
                  "borderBottomRightRadius": "10px",
                  "borderTopLeftRadius": "10px",
                  "borderTopRightRadius": "10px",
                  "height": "20px",
                  "opacity": 0,
                  "transform": "translateX(0px) translateY(0px)",
                  "width": "20px",
                }
              }
            />
            <div
              className="css-view-1dbjc4n r-position-u8s1d"
              onClickCapture={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0.00)",
                  "bottom": "0px",
                  "left": "0px",
                  "marginBottom": "-20px",
                  "marginLeft": "-20px",
                  "marginRight": "-20px",
                  "marginTop": "-20px",
                  "right": "0px",
                  "top": "0px",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "marginTop": "2%",
          }
        }
      >
        Logout
      </div>
    </div>
  </div>,
]
`;
